netaddr.go

```diff
@@ -778,6 +780,45 @@ func (p IPPrefix) Contains(addr IP) bool {
 	return true
 }
 
+// Overlaps reports whether p and o overlap at all.
+//
+// If p and o are of different address families or either have a zero
+// IP, it reports false.
+//
+// If either has a Bits of zero, it returns true.
+func (p IPPrefix) Overlaps(o IPPrefix) bool {
+	if p.IP.IsZero() || o.IP.IsZero() {
+		return false
+	}
+	if p == o {
+		return true
+	}
+	if p.IP.Is4() != o.IP.Is4() {
```

Comment:
It might be worth a comment mentioning what you intend to happen here with IPv6-mapped IPv4 addresses.

Reply:
Right! Will do.